<?xml version="1.0" ?>
<!-- =================================================================================== -->
<!-- |    This document was autogenerated by xacro from TypeA.xacro                    | -->
<!-- |    EDITING THIS FILE BY HAND IS NOT RECOMMENDED                                 | -->
<!-- =================================================================================== -->
<robot name="awesome-o" xmlns:xacro="http://www.ros.org/wiki/xacro">
  <link name="base_link">
    <visual>
      <geometry>
        <box size="0.11 0.05 0.12"/>
      </geometry>
    </visual>
  </link>
  <link name="head">
    <visual>
      <geometry>
        <box size="0.06 0.06 0.04"/>
      </geometry>
      <origin rpy="0 0 0" xyz="0.0 0 0.05"/>
      <material name="yellow">
      <color rgba=".8 .8 0 1"/>
      </material>
    </visual>
  </link>
  <joint name="base_to_head" type="revolute">
    <parent link="base_link"/>
    <child link="head"/>
    <origin rpy="0 0 0" xyz="0.0 0 0.05"/>
    <axis xyz="0 0 1"/>
    <limit effort="1000.0" lower="-3.14" upper="3.14" velocity="0.5"/>
  </joint>
  <link name="right_shoulder">
    <visual>
      <geometry>
        <box size="0.03 0.04 0.03"/>
      </geometry>
      <origin rpy="0 0 0" xyz="0.0 0 0.00"/>
      <material name="blue">
        <color rgba="0 0 .8 1"/>
      </material>
    </visual>
  </link>
  <joint name="base_to_right_shoulder" type="revolute">
    <parent link="base_link"/>
    <child link="right_shoulder"/>
    <origin rpy="1.5 0 0" xyz="0.07 0 0.03"/>
    <axis xyz="1 0 0"/>
    <limit effort="1000.0" lower="-3.14" upper="3.14" velocity="0.5"/>
  </joint>
  <link name="right_upper_arm">
    <visual>
      <geometry>
        <!--mesh filename="file:///home/local/AX12.dae" scale="0.0001 0.0001 0.0001"/-->
        <box size="0.03 0.04 0.09"/>
      </geometry>
      <material name="green">
        <color rgba="0 .8 0 1"/>
      </material>
      <origin rpy="0 0 0" xyz="0.00 0 -0.045"/>
    </visual>
  </link>
  <joint name="right_shoulder_to_upper" type="revolute">
    <parent link="right_shoulder"/>
    <child link="right_upper_arm"/>
    <origin rpy="0 -1.5 0" xyz="0.00 0 0.0"/>
    <axis xyz="0 -1 0"/>
    <limit effort="1000.0" lower="-3.14" upper="3.14" velocity="0.5"/>
  </joint>
  <link name="right_forearm">
    <visual>
      <geometry>
        <!--mesh filename="file:///home/local/AX12.dae" scale="0.0001 0.0001 0.0001"/-->
        <box size="0.03 0.04 0.09"/>
      </geometry>
      <material name="blue"/>
      <origin rpy="0 0 0" xyz="0.00 0 -0.045"/>
    </visual>
  </link>
  <joint name="right_elbow" type="revolute">
    <parent link="right_upper_arm"/>
    <child link="right_forearm"/>
    <origin rpy="0 0 0" xyz="0.00 0 -0.08"/>
    <axis xyz="0 -1 0"/>
    <limit effort="1000.0" lower="-3.14" upper="3.14" velocity="0.5"/>
  </joint>
  <link name="left_shoulder">
    <visual>
      <geometry>
        <box size="0.03 0.04 0.03"/>
      </geometry>
      <origin rpy="0 0 0" xyz="0.0 0 0.00"/>
      <material name="blue">
        <color rgba="0 0 .8 1"/>
      </material>
    </visual>
  </link>
  <joint name="base_to_left_shoulder" type="revolute">
    <parent link="base_link"/>
    <child link="left_shoulder"/>
    <origin rpy="1.5 0 0" xyz="-0.07 0 0.03"/>
    <axis xyz="-1 0 0"/>
    <limit effort="1000.0" lower="-3.14" upper="3.14" velocity="0.5"/>
  </joint>
  <link name="left_upper_arm">
    <visual>
      <geometry>
        <!--mesh filename="file:///home/local/AX12.dae" scale="0.0001 0.0001 0.0001"/-->
        <box size="0.03 0.04 0.09"/>
      </geometry>
      <material name="green">
        <color rgba="0 .8 0 1"/>
      </material>
      <origin rpy="0 0 0" xyz="0.00 0 -0.045"/>
    </visual>
  </link>
  <joint name="left_shoulder_to_upper" type="revolute">
    <parent link="left_shoulder"/>
    <child link="left_upper_arm"/>
    <origin rpy="0 1.5 0" xyz="0.00 0 0.0"/>
    <axis xyz="0 -1 0"/>
    <limit effort="1000.0" lower="-3.14" upper="3.14" velocity="0.5"/>
  </joint>
  <link name="left_forearm">
    <visual>
      <geometry>
        <!--mesh filename="file:///home/local/AX12.dae" scale="0.0001 0.0001 0.0001"/-->
        <box size="0.03 0.04 0.09"/>
      </geometry>
      <material name="blue"/>
      <origin rpy="0 0 0" xyz="0.00 0 -0.045"/>
    </visual>
  </link>
  <joint name="left_elbow" type="revolute">
    <parent link="left_upper_arm"/>
    <child link="left_forearm"/>
    <origin rpy="0 0 0" xyz="0.00 0 -0.08"/>
    <axis xyz="0 -1 0"/>
    <limit effort="1000.0" lower="-3.14" upper="3.14" velocity="0.5"/>
  </joint>
  <link name="right_hip">
    <visual>
      <geometry>
        <box size="0.03 0.04 0.03"/>
      </geometry>
      <origin rpy="0 0 0" xyz="0.0 0 0.00"/>
      <material name="blue">
        <color rgba="0 0 .8 1"/>
      </material>
    </visual>
  </link>
  <joint name="base_to_right_hip" type="revolute">
    <parent link="base_link"/>
    <child link="right_hip"/>
    <origin rpy="0 0 -0.7" xyz="0.03 0 -0.06"/>
    <axis xyz="0 0 -1"/>
    <limit effort="1000.0" lower="-3.14" upper="3.14" velocity="0.5"/>
  </joint>
  <link name="right_hip_split">
    <visual>
      <geometry>
        <box size="0.03 0.04 0.03"/>
      </geometry>
      <material name="red">
        <color rgba="1 .0 .0 1"/>
      </material>
      <origin rpy="0 0 0" xyz="0.00 0 -0.045"/>
    </visual>
  </link>
  <joint name="right_hip_to_split" type="revolute">
    <parent link="right_hip"/>
    <child link="right_hip_split"/>
    <origin rpy="0 0 0" xyz="0.00 0 0.0"/>
    <axis xyz="0 1 0"/>
    <limit effort="1000.0" lower="-3.14" upper="3.14" velocity="0.5"/>
  </joint>
  <link name="right_upper_leg">
    <visual>
      <geometry>
        <box size="0.03 0.04 0.09"/>
      </geometry>
      <material name="green"/>
      <origin rpy="0 0 0" xyz="0.00 0 -0.045"/>
    </visual>
  </link>
  <joint name="right_hip_split_to_upper" type="revolute">
    <parent link="right_hip_split"/>
    <child link="right_upper_leg"/>
    <origin rpy="0 0 0" xyz="0.00 0 0.0"/>
    <axis xyz="-1.0 0 0"/>
    <limit effort="1000.0" lower="-3.14" upper="3.14" velocity="0.5"/>
  </joint>
  <link name="right_calf">
    <visual>
      <geometry>
        <!--mesh filename="file:///home/local/AX12.dae" scale="0.0001 0.0001 0.0001"/-->
        <box size="0.03 0.04 0.09"/>
      </geometry>
      <material name="blue"/>
      <origin rpy="0 0 0" xyz="0.00 0 -0.045"/>
    </visual>
  </link>
  <joint name="right_knee" type="revolute">
    <parent link="right_upper_leg"/>
    <child link="right_calf"/>
    <origin rpy="0 0 0" xyz="0.00 0 -0.08"/>
    <axis xyz="1.0 0 0"/>
    <limit effort="1000.0" lower="-3.14" upper="3.14" velocity="0.5"/>
  </joint>
  <link name="right_foot">
    <visual>
      <geometry>
        <box size="0.05 0.075 0.05"/>
      </geometry>
      <material name="green"/>
      <origin rpy="0 0 0" xyz="0.00 0 -0.045"/>
    </visual>
  </link>
  <joint name="right_ankle" type="revolute">
    <parent link="right_calf"/>
    <child link="right_foot"/>
    <origin rpy="0 0 0" xyz="0.00 0 -0.045"/>
    <axis xyz="1 0 0"/>
    <limit effort="1000.0" lower="-3.14" upper="3.14" velocity="0.5"/>
  </joint>
  <link name="right_foot_pad">
    <visual>
      <geometry>
        <box size="0.06 0.10 0.005"/>
      </geometry>
      <material name="blue"/>
      <origin rpy="0 0 0" xyz="0.00 0 -0.045"/>
    </visual>
  </link>
  <joint name="right_ankle_roll" type="revolute">
    <parent link="right_foot"/>
    <child link="right_foot_pad"/>
    <origin rpy="0 0 0" xyz="0.00 0 -0.03"/>
    <axis xyz="0 -1 0"/>
    <limit effort="1000.0" lower="-3.14" upper="3.14" velocity="0.5"/>
  </joint>
  <link name="left_hip">
    <visual>
      <geometry>
        <box size="0.03 0.04 0.03"/>
      </geometry>
      <origin rpy="0 0 0" xyz="0.0 0 0.00"/>
      <material name="blue">
        <color rgba="0 0 .8 1"/>
      </material>
    </visual>
  </link>
  <joint name="base_to_left_hip" type="revolute">
    <parent link="base_link"/>
    <child link="left_hip"/>
    <origin rpy="0 0 0.7" xyz="-0.03 0 -0.06"/>
    <axis xyz="0 0 -1"/>
    <limit effort="1000.0" lower="-3.14" upper="3.14" velocity="0.5"/>
  </joint>
  <link name="left_hip_split">
    <visual>
      <geometry>
        <box size="0.03 0.04 0.03"/>
      </geometry>
      <material name="red">
        <color rgba="1 .0 .0 1"/>
      </material>
      <origin rpy="0 0 0" xyz="0.00 0 -0.045"/>
    </visual>
  </link>
  <joint name="left_hip_to_split" type="revolute">
    <parent link="left_hip"/>
    <child link="left_hip_split"/>
    <origin rpy="0 0 0" xyz="0.00 0 0.0"/>
    <axis xyz="0 1 0"/>
    <limit effort="1000.0" lower="-3.14" upper="3.14" velocity="0.5"/>
  </joint>
  <link name="left_upper_leg">
    <visual>
      <geometry>
        <box size="0.03 0.04 0.09"/>
      </geometry>
      <material name="green"/>
      <origin rpy="0 0 0" xyz="0.00 0 -0.045"/>
    </visual>
  </link>
  <joint name="left_hip_split_to_upper" type="revolute">
    <parent link="left_hip_split"/>
    <child link="left_upper_leg"/>
    <origin rpy="0 0 0" xyz="0.00 0 0.0"/>
    <axis xyz="1.0 0 0"/>
    <limit effort="1000.0" lower="-3.14" upper="3.14" velocity="0.5"/>
  </joint>
  <link name="left_calf">
    <visual>
      <geometry>
        <!--mesh filename="file:///home/local/AX12.dae" scale="0.0001 0.0001 0.0001"/-->
        <box size="0.03 0.04 0.09"/>
      </geometry>
      <material name="blue"/>
      <origin rpy="0 0 0" xyz="0.00 0 -0.045"/>
    </visual>
  </link>
  <joint name="left_knee" type="revolute">
    <parent link="left_upper_leg"/>
    <child link="left_calf"/>
    <origin rpy="0 0 0" xyz="0.00 0 -0.08"/>
    <axis xyz="-1.0 0 0"/>
    <limit effort="1000.0" lower="-3.14" upper="3.14" velocity="0.5"/>
  </joint>
  <link name="left_foot">
    <visual>
      <geometry>
        <box size="0.05 0.075 0.05"/>
      </geometry>
      <material name="green"/>
      <origin rpy="0 0 0" xyz="0.00 0 -0.045"/>
    </visual>
  </link>
  <joint name="left_ankle" type="revolute">
    <parent link="left_calf"/>
    <child link="left_foot"/>
    <origin rpy="0 0 0" xyz="0.00 0 -0.045"/>
    <axis xyz="-1 0 0"/>
    <limit effort="1000.0" lower="-3.14" upper="3.14" velocity="0.5"/>
  </joint>
  <link name="left_foot_pad">
    <visual>
      <geometry>
        <box size="0.06 0.10 0.005"/>
      </geometry>
      <material name="blue"/>
      <origin rpy="0 0 0" xyz="0.00 0 -0.045"/>
    </visual>
  </link>
  <joint name="left_ankle_roll" type="revolute">
    <parent link="left_foot"/>
    <child link="left_foot_pad"/>
    <origin rpy="0 0 0" xyz="0.00 0 -0.03"/>
    <axis xyz="0 -1 0"/>
    <limit effort="1000.0" lower="-3.14" upper="3.14" velocity="0.5"/>
  </joint>
</robot>
